src/DevBetterWeb.Vimeo/Services/VideoServices/CreateAnimatedThumbnailsService.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using DevBetterWeb.Vimeo.Models;
+using Microsoft.Extensions.Logging;
+
+namespace DevBetterWeb.Vimeo.Services.VideoServices;
+
+public class CreateAnimatedThumbnailsService
+{
+  private readonly GetAnimatedThumbnailService _getAnimatedThumbnailService;
+  private readonly GetStatusAnimatedThumbnailService _getStatusAnimatedThumbnailService;
+  private readonly AddAnimatedThumbnailsToVideoService _addAnimatedThumbnailsToVideoService;
+  private readonly GetVideoService _getVideoService;
+  private readonly ILogger<CreateAnimatedThumbnailsService> _logger;
+
+  public CreateAnimatedThumbnailsService(
+    GetAnimatedThumbnailService getAnimatedThumbnailService, 
+    GetStatusAnimatedThumbnailService getStatusAnimatedThumbnailService, 
+    AddAnimatedThumbnailsToVideoService addAnimatedThumbnailsToVideoService, 
+    GetVideoService getVideoService, 
+    ILogger<CreateAnimatedThumbnailsService> logger)
+  {
+    _getAnimatedThumbnailService = getAnimatedThumbnailService;
+    _getStatusAnimatedThumbnailService = getStatusAnimatedThumbnailService;
+    _addAnimatedThumbnailsToVideoService = addAnimatedThumbnailsToVideoService;
+    _getVideoService = getVideoService;
+    _logger = logger;
+  }
+
+  public async Task<AnimatedThumbnailsResponse> ExecuteAsync(long videoId, CancellationToken cancellationToken = default)
+  {
+    return await CreateAnimatedThumbnails(videoId);
+  }
+
+  private async Task<AnimatedThumbnailsResponse> CreateAnimatedThumbnails(long videoId)
+  {
+    Video video = new Video();
+    while (video == null || video.Status != "available")
+    {
+      Thread.Sleep(20 * 1000);
```

Comment:
Why do we seem to be sleeping before we even do anything?

Reply:
if there is one call happen before this